sos/report/plugins/tigervnc.py

```diff
@@ -12,17 +12,33 @@
 
 
 class TigerVNC(Plugin, RedHatPlugin):
+    """
+    This plugin gathers information for VNC servers provided by the tigervnc
+    package. This is explicitly for server-side collections, not clients.
+
+    By default, this plugin will capture the contents of /etc/tigervnc, which
+    may include usernames. If usernames are sensitive information for end
+    users of sos, consider using the `--clean` option to obfuscate these
+    names.
+    """
 
     short_desc = 'TigerVNC server configuration'
     plugin_name = 'tigervnc'
     packages = ('tigervnc-server',)
 
     def setup(self):
-        self.add_copy_spec([
-            '/etc/tigervnc/vncserver-config-defaults',
-            '/etc/tigervnc/vncserver-config-mandatory',
-            '/etc/tigervnc/vncserver.users'
-        ])
+        self.add_copy_spec('/etc/tigervnc/')
+
+        # service names are 'vncserver@$port' where $port is :1,, :2, etc...
+        # however they are not reported via list-unit-files, only list-units
+        vncs = self.exec_cmd(
+            'systemctl list-units --type=service --no-legend vncserver*'
+        )
+        if vncs['status'] == 0:
+            for serv in vncs['output'].splitlines():
+                vnc = serv.split()
+                self.add_service_status(vnc[0])
+                self.add_journal(vnc[0])
```

Comment:
Can't the `systemctl list-units ..` command output contain also a blank line (I really dont know this)? As `''.split()[0]` raises index out of range error.

Reply:
It _shouldn't_ with `--no-legend`, but it would still be theoretically possible, so I'll add a catch there. Good idea.